packages/scrollregion/src/useScrollRegion.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+// Scroll regions with dynamic layouts should use the dependencies prop.
+// The hook re-calculates the tab-index its logic when the dependencies change.
+
+import { useState, useMemo, useEffect } from 'react';
+import debounce from 'lodash.debounce';
+
+export interface IUseScrollRegion {
+  /** A [ref](https://reactjs.org/docs/refs-and-the-dom.html) pointing to the scroll region's container element */
+  containerRef: React.RefObject<HTMLElement>;
+  /** A value that determines if the scroll region tab index should be recalculated */
+  dependency?: any;
```

Comment:
I'm sure you thought about the name, but `dependency` feels a bit obscure to me. In other places, we tend to use `xxxProps`. Would something like `contentProps` or similar be more self-referential here?

Although the use of `xxxProps` throughout is different (consider `backdropProps` on `DrawerModal`). Either way, the name feels like it could be improved. YMMV.

Reply:
Would it be appropriate to append `Props` to the variable name when the value is not used like `backdropProps` or  `buttonProps`? For example, the value is not spread onto a React element as props but rather passed to a `useEffect` dependency array. The value can also be a non-reference value (a type of `number`, `boolean`, `string`, etc).

The naming of this one was difficult. I can see why it may feel a bit obscure to others. What do you think about the variable name `contentChange`? I think it could better describe the potential values. A potential value could be a boolean such as `hasOverflow` or a memoized array/object such as `[hasOverflow, children]`.

Reply:
You're right about `Props` – it's not consistent.

I like `contentChange` less than `dependency`. I guess we can just leave as-is unless you uncover a thought that feels more standard/consistent with other APIs.

Reply:
It's actually helpful to note that the React documentation does term these `dependencies`. https://reactjs.org/docs/hooks-reference.html#usememo.

Reply:
Since the value can be non-reference value I decided to keep the name `dependency`. Let me know if you think that should change.

Alternatively, the value could be renamed from `dependency` to `dependencies` and the type can be updated from `any` to `any[]`. Consumers would always need to memoize (even when a single, non-reference value is used). Maybe this is a simpler API?

```jsx
useScrollRegion({ containerRef, dependency: hasOverflow })
```

vs

```jsx
const dependencies = useMemo(() => {
  return [hasOverflow]
}, [hasOverflow])

useScrollRegion({ containerRef, dependencies })
```